### PR TITLE
add metric to check if validator is in next sync commitee

### PR DESCRIPTION
**Pull request by bharath-123**
**What type of PR is this?**
Enhancement

**What does this PR do? Why is it needed?**

This PR adds a prometheus metric to check if a validator is in the next sync commitee. We currently already have a metric to check if a validator is in the current sync commitee

This issue also resolves a bug where the metric `ValidatorInSyncCommitteeGaugeVec` won't be set to 0 if the validator is not in the current sync commitee. As validators are shuffled b/w sync committees every 27 hours, we will have to reset this metric to 0, if the validator is moved out of the sync commitee.

**Which issues(s) does this PR fix?**

Fixes https://github.com/prysmaticlabs/prysm/issues/12269

**Other notes for review**

### CODE REVIEW

**Review by nisdas — DISMISSED** [review]

**Comment by bharath-123** [comment]
@nisdas have rebased the PR and fixed the typo

**Review by nisdas — DISMISSED** [review]

File: validator/client/validator.go [hunk 1]
```diff
@@ -926,6 +929,23 @@ func (v *validator) logDuties(slot primitives.Slot, duties []*ethpb.DutiesRespon
 			}
 		}
 	}
+	for _, duty := range nextEpochDuties {
+		// for the next epoch, currently we are only interested in whether the validator is in the next sync committee or not
+		validatorNotTruncatedKey := fmt.Sprintf("%#x", duty.PublicKey)
+
+		// Only interested in validators who are attesting/proposing.
+		// Note that SLASHING validators will have duties but their results are ignored by the network so we don't bother with them.
```
Comment (rkapka) [inline validator/client/validator.go:937]:
```suggestion
		// Note that slashed validators will have duties but their results are ignored by the network so we don't bother with them.
```

**Review by rkapka — APPROVED** [review]

File: validator/client/metrics.go [hunk 2]
```diff
@@ -201,6 +201,17 @@ var (
 			"pubkey",
 		},
 	)
+	// ValidatorInNextSyncCommitteeGaugeVec used to track whether validator is in the next sync committee.
+	ValidatorInNextSyncCommitteeGaugeVec = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "validator",
+			Name:      "in_next_sync_committee",
+			Help:      "validator next sync committee.New in Altair hardfork",
```
Comment (saolyn) [inline validator/client/metrics.go:209]: Super tiny nitpick, same for line 198:
```suggestion
			Help:      "validator next sync committee. New in Altair hardfork",
```

**Comment by bharath-123** [comment]
Will rebase this PR and push again

**Comment by bharath-123** [comment]
hey @nisdas can this be merged? would like to resolve any thing that is blocking merging

**Comment by bharath-123** [comment]
Done

**Review by nisdas — APPROVED** [review]

**Pull request merged**